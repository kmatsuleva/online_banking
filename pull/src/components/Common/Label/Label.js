import React from "react";
import PropTypes from 'prop-types';     

const Label = ({ text, style }) => (
  <label style= {style}> { text } </label>
);

Label.propTypes = {
  text: PropTypes.string,
  style: PropTypes.object    
};


Label.defaultProps = {
  style: {
      fontSize: 18,
      color: "#4a4a4a",
      fontFamily: "Roboto, Arial, sans-serif"
  }
}


export default Label;
