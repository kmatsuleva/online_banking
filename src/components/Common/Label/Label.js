import React from "react";
import PropTypes from 'prop-types';     

const Label = ({ text, className }) => (
  <label className= {className}> { text } </label>
);

Label.propTypes = {
  text: PropTypes.string,
  style: PropTypes.object    
};



export default Label;
