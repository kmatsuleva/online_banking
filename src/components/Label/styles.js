const styles = {
    label: {
        fontSize: 18,
        color: "#4a4a4a",
        fontFamily: "Roboto, Arial, sans-serif",
        marginBottom: 8
    }
}

export default styles
