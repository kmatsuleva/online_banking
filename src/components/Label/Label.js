import React from "react";
import styles from "./styles.js"

const Label = ({ labelValue }) => (
  <div>
      <label style={styles.label} > {labelValue} </label>
  </div>
);

export default Label;
 