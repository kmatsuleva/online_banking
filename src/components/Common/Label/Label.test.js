
import React from 'react'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Label from "./Label"

Enzyme.configure({ adapter: new Adapter() })

describe("<Label />", () => {

    it('renders without crashing', () => {
        shallow(<Label />);
    });

    it('should render correctly', () => {
        const output = shallow(
        <Label text="mockText" />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});