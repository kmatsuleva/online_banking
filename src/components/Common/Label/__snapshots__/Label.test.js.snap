// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render correctly 1`] = `
<label>
   
  mockText
   
</label>
`;
